### ISSUE TITLE
Couldn't reproduce the results (either from scratch or from pre-trained)

### ISSUE THREAD

**Issue by KnightZhang625** [chen-judge/UniGeo#1]
Hello,

I tried to train from scratch (loaded the [pretrained.pth](https://drive.google.com/drive/folders/1NifdHLJe5U08u2Zb1sWL6C-8krpV2z2O?usp=share_link)). In addition, I also evaluated by loading your [geoformer.pth](https://drive.google.com/drive/folders/1NifdHLJe5U08u2Zb1sWL6C-8krpV2z2O?usp=share_link)). However, the results are not the similar to the reported in the papers, the following are the results from the stdout:

- loaded the pretrained.pth, and trained from scratch:
<img width="644" alt="Screen Shot 2022-12-20 at 13 15 29" src="https://user-images.githubusercontent.com/36263466/208675926-8ce05736-21d3-40eb-8fad-788b2665098f.png">

- loaded the geoformer.pth, and test directly:
<img width="642" alt="Screen Shot 2022-12-20 at 13 16 17" src="https://user-images.githubusercontent.com/36263466/208676061-22c358d6-0403-41f3-b5a7-0dd288b7dd38.png">

As you can see, the numbers following the Acc for either "Calculation" or "Proving" fall behind the reported numbers in the paper. And I only modified the bash scripts to make it train on one GPU (A100), leaving other parts unchanged. Could you please tell me whether there is anything I need to pay attention to, so that I could reproduce the reported numbers. 

Thanks!

**Comment by chen-judge** [comment]
Hi,

The beam size should be set to 10. I guess you set it to 1.
Please refer to evaluate.sh and check: `--num_beams 10`

**Comment by KnightZhang625** [comment]
Hi Chen,

Thanks for your suggestions! I used the `evaluate.sh `and get the results close to the paper.
<img width="863" alt="Screen Shot 2022-12-21 at 14 41 32" src="https://user-images.githubusercontent.com/36263466/208931695-7e373cc2-642f-4573-aa6d-dc8b7814ce8f.png">

But I still have one question: When I add the `--test_only` and modify the `--num_beams 10` for `train.sh`, why the results show the accuracy is 0.0 for both types? From my understanding, `--test_only` forces to execute the trainer.test(), which is the similar function when using `evaluate.sh `. 

Thanks!

**Comment by chen-judge** [comment]
Have you loaded the correct checkpoints? I think you might have missed a few details.
